web/src/main/java/org/cbioportal/web/parameter/GeneFilter.java

```diff
@@ -0,0 +1,125 @@
+package org.cbioportal.web.parameter;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.AssertTrue;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+public class GeneFilter implements Serializable {
+
+    private Set<String> molecularProfileIds;
+    private List<List<String>> geneQueries;
+
+    private final String GENE_QUERY_PATTERN = "^(\\w+)[\\s]*?(?:\\:(?:[\\s]*(?:(AMP)|(HOMDEL))\\b)+)?$";
+
+    public enum CNAType {
+        AMP("AMP", 2), GAIN("GAIN", 1), HOMDEL("HOMDEL", -2), HETLOSS("HETLOSS", -1);
+
+        private final String name;
+        private final Integer number;
+
+        private CNAType(String name, Integer number) {
+            this.name = name;
+            this.number = number;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public Integer getNumber() {
+            return number;
+        }
+    }
+
+    public class SingleGeneQuery implements Serializable {
+        private String hugoGeneSymbol;
+        private List<CNAType> alterations;
+
+        public String getHugoGeneSymbol() {
+            return hugoGeneSymbol;
+        }
+
+        public void setHugoGeneSymbol(String hugoGeneSymbol) {
+            this.hugoGeneSymbol = hugoGeneSymbol;
+        }
+
+        public List<CNAType> getAlterations() {
+            return alterations;
+        }
+
+        public void setAlterations(List<CNAType> alterations) {
+            this.alterations = alterations;
+        }
+    }
+
+    @AssertTrue
+    private boolean isValid() {
```

Comment:
How do we use this method?

Reply:
it is invoked when the object is set. If invalid object is passed through the api it returns error